### PR TITLE
fix: Filter chips lacks width for longer values

### LINKED ISSUE #7018: [bug 🐛 ]:  Filter chips lacks width for longer values
## Bug Description

- In mobile viewport, Filter chips lacks sufficient width which causes them to break spaces on longer values

## Before Behaviour
Screenshot 2024-09-13 at 4 52 43 PM

## Expected behavior

Approach 1 - 

https://github.com/user-attachments/assets/ee12fe64-ed0c-4c35-8436-f56c04722f59

Approach 2- 

Screenshot 2024-09-13 at 5 00 28 PM

### PR DESCRIPTION
**Pull request by harshit078**
## Description

Closes #7018

- When given longer values, filter chips break-spaces and lack sufficient width
- As a result, a design overhaul is given to `StyledBar` and `StyledChipcontainer` components.

## Before

- on Desktop
Screenshot 2024-09-15 at 1 19 00 AM

- On mobile viewport
Screenshot 2024-09-15 at 1 19 26 AM

Screenshot 2024-09-15 at 1 19 54 AM

## After

- On desktop
Screenshot 2024-09-15 at 1 20 41 AM

- On mobile viewport
Screenshot 2024-09-15 at 1 25 38 AM

https://github.com/user-attachments/assets/0b4ff758-3b6e-4bd5-8247-6b096fa7d1c0

### CODE REVIEW

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 1]
```diff
@@ -29,21 +29,23 @@ const StyledBar = styled.div`
   border-top: 1px solid ${({ theme }) => theme.border.color.light};
   display: flex;
   flex-direction: row;
-  min-height: 32px;
+  align-items: center;
   justify-content: space-between;
+  position: relative;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:34]: position: relative does not seem necessary, let's remove it

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 2]
```diff
@@ -29,21 +29,23 @@ const StyledBar = styled.div`
...
+  padding: ${({ theme }) => theme.spacing(1)};
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:35]: this is not matching the Figma, on the figma we only have a padding-top and padding-bottom, no left and right

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 3]
```diff
@@ -29,21 +29,23 @@ const StyledBar = styled.div`
...
 
 const StyledChipcontainer = styled.div`
   align-items: center;
   display: flex;
   flex-direction: row;
   gap: ${({ theme }) => theme.spacing(1)};
-  min-height: 32px;
-  margin-left: ${({ theme }) => theme.spacing(2)};
-  flex-wrap: wrap;
+  overflow-x: scroll;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:45]: I'm not sure about this scroll behavior, it's very difficult to use. I think we want they to pile vertically
@Bonapara could confirm

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 4]
```diff
@@ -29,21 +29,23 @@ const StyledBar = styled.div`
...
+  padding-top: ${({ theme }) => theme.spacing(1)};
+  white-space: nowrap;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:47]: let's modify the SortOrFilterChip itself and add this white-space: no-wrap property there

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 5]
```diff
@@ -29,21 +29,23 @@ const StyledBar = styled.div`
...
 const StyledChipcontainer = styled.div`
   align-items: center;
   display: flex;
   flex-direction: row;
   gap: ${({ theme }) => theme.spacing(1)};
-  min-height: 32px;
-  margin-left: ${({ theme }) => theme.spacing(2)};
-  flex-wrap: wrap;
+  overflow-x: scroll;
+  padding-top: ${({ theme }) => theme.spacing(1)};
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:46]: let's also add the padding-bottom to be consistent

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx (hunk 3 above)
Comment (charlesBochet) [reply packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:45]: let's remove the scroll

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 6]
```diff
@@ -52,14 +54,8 @@ const StyledCancelButton = styled.button`
...
-  margin-left: auto;
-  margin-right: ${({ theme }) => theme.spacing(2)};
-  padding: ${(props) => {
-    const horiz = props.theme.spacing(2);
-    const vert = props.theme.spacing(1);
-    return `${vert} ${horiz} ${vert} ${horiz}`;
-  }};
-  user-select: none;
+  padding: ${({ theme }) => `${theme.spacing(1)} ${theme.spacing(2)}`};
+  z-index: 10;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:58]: not sure why we need it too

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 7]
```diff
@@ -52,14 +54,8 @@ const StyledCancelButton = styled.button`
...
   font-weight: ${({ theme }) => theme.font.weight.medium};
-  margin-left: auto;
-  margin-right: ${({ theme }) => theme.spacing(2)};
-  padding: ${(props) => {
-    const horiz = props.theme.spacing(2);
-    const vert = props.theme.spacing(1);
-    return `${vert} ${horiz} ${vert} ${horiz}`;
-  }};
-  user-select: none;
+  padding: ${({ theme }) => `${theme.spacing(1)} ${theme.spacing(2)}`};
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:57]: +1

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 8]
```diff
@@ -52,14 +54,8 @@ const StyledCancelButton = styled.button`
...
   cursor: pointer;
   font-weight: ${({ theme }) => theme.font.weight.medium};
-  margin-left: auto;
-  margin-right: ${({ theme }) => theme.spacing(2)};
-  padding: ${(props) => {
-    const horiz = props.theme.spacing(2);
-    const vert = props.theme.spacing(1);
-    return `${vert} ${horiz} ${vert} ${horiz}`;
-  }};
-  user-select: none;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:62]: why remove user-select and margin-right? they are useful!

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx [hunk 9]
```diff
@@ -68,8 +64,10 @@ const StyledCancelButton = styled.button`
 `;
 
 const StyledFilterContainer = styled.div`
-  align-items: center;
   display: flex;
+  align-items: center;
+  flex: 1;
+  overflow-x: hidden;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:70]: we need item to stack

**Review by charlesBochet — COMMENTED** [review]
@harshit078 Thanks for the issue and PR :)

However, I'm not sure about these changes:
- I think the initial issue about avoiding wrapping within chip should be done in SortOrFilterChip direclty
- about the scrolling horizontal (which does not match your PR screenshot btw) I don't think it improves the UX, we want to keep piling the filters and sorts vertically
- I see several additions (z-index, position) that does not seem effective
- I see changes in margin and padding that do not match the figma: https://www.figma.com/design/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=15351-156515&node-type=frame&t=XctmAQ2oAV3UJRoV-0

Could you rework it?
Also make sure that it looks good on (desktop / mobile) x (with sort / without sort) x (with filter / without filter)

Thank you!

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx (hunk 3 above)
Comment (harshit078) [reply packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:45]: @charlesBochet, In the figma designs and the video in #7018 the filter chips scroll vertically behind the Reset and Save as new view state. If we remove this scroll then how should we approach it ?

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx (hunk 9 above)
Comment (harshit078) [reply packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx:70]: On the dev environment, filters stack on each other in column manner. But in figma they are stacking row wise and having a scroll. Is column stack behaviour we need to follow ?

File: packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx (hunk 3 above)
Comment (charlesBochet) [reply packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx]: oh ok! I missed that, I feel this is not easy to use and need to be discussed again with Bonapara for small screens but we could merge the PR

Comment (harshit078) [reply packages/twenty-front/src/modules/views/components/ViewBarDetails.tsx]: @charlesBochet, I agree to your point. To overcome this, can we can do a tooltip or information badge for all mobile viewports as a result when user first time applies filters, they can learn ?

**Comment by lucasbordeau** [comment]
Works like that, merging as it corresponds to the issue.

**Review by lucasbordeau — APPROVED** [review]

**Pull request merged**